### ISSUE TITLE
FR: Loading more than one sample into sample player (with number slider or equivalent to change playing sample)

### ISSUE THREAD

**Issue by Lykdraft** [kmatheussen/radium#643]
Another one that sound a bit far-fetched, but is very interesting to me would be the ability
to load multiple samples into the sample player and with the help of some kind of slider/number box
(which can  be automated) you can select which sample actually should play back the notes.

This allows for a lot of things, for example you could load multiple kicks into one sample player that you like and then still change your fav kick mid- or even end-production. (Basically collecting more than one good sound and going through your "fav" collection afterwards to pick the 
best one for the song. 

Second, this allows for **very** high profile production techniques 
that are otherwise pretty difficult to accomplish. For example: 
- Playing a melody and changing the sample while the melody is playing.
 - Playing a loop and automate between different loaded loops, which then creates some kind of a "superloop" or "loop-mashup" 
You can get very very fancy with this. D'n'B and Dubstep for example can't be done without this. 

When I am right, this would not require very much to add, because all the time there is still only one sample playing. Radiums Sample Player would just need the ability to change the sample on the fly and load more than one sample at once into some kind of memory array. 
No need for multi-sampler stuff like keygroups or zones , deeper editing pages or whatever
but still quite a huge improvement of the Players possibilities. 

Again this may or may not be easy to realize, depending on the flexibility and backend of the sample player code itself. But I am really interested in this because chances are high that this might be actually more on the easy side. And this is another one that is highly unusual but one of the top-features for a lot of people in Ableton Live for example.

**Comment by kmatheussen** [comment]
I think this feature would be simple to implement too. It should be enough having an array of 256 slots for samples, and then use the slot pointed to by the effect "slot number", even when loading new sample. (it's a little bit more work than that, but not much).

Regarding zones, etc. a lot of things are actually supported already, since the sample player can play soundfonts, but there isn't interface for those things.

**Comment by Lykdraft** [comment]
> It should be enough having an array of 256 slots for samples, and then use the slot pointed to by the effect "slot number", even when loading new sample.

***Very*** good. That's more than enough that's needed for this.

> Regarding zones, etc. a lot of things are actually supported already, since the sample player can play soundfonts, but there isn't interface for those things.

Oh, interesting. But for the techniques in question this really is unneccessary for now IMHO.

**Comment by kmatheussen** [comment]
Hmm, it's probably necessary to have a small crossfade when switching sample while playing. If not there will be clicks. Now it starts to become complicated.

**Comment by Lykdraft** [comment]
Not if the Attack/release settings are getting handled identically by all samples like if they were one and the same sample... There should be no clicks.. 

_(A bit the same like when I automate sample start. There also would be heavy clicks when I don't adjust attack and release times. If I do, smooth as silk... No clicks whatsoever.)_

**Comment by kmatheussen** [comment]
Isn't the point about this that the sample is switched while a note is playing?

**Comment by Lykdraft** [comment]
No. Think of it that way like when you automate the sample start. The note triggers 
the sample at that specific point in time. You can't change the sample position while a note is playing. 
You change the sample start and the next note triggers the sample 
at the new specified sample position. 

With multiple samples it is the same thing. The note triggers the specified sample number.
Then you change the sample number, and the next note triggers the specified ***new*** sample number. 

When you would like to have a "crossfade" effect, then you just enter a large amount of notes (like 32nd/or64th) in rapid succesion on the same key, like C4 for example, so that the sample point (or sample number) gets triggered faster/more often, which then sounds like some kind of crossfade effect. At least this is how I do it. But the note is still the trigger.

**Comment by Lykdraft** [comment]
Stumbled on this old thread. Just want to add that this would open the possibility for round-robin... 

(FYI: http://www.musicradar.com/tuition/tech/how-to-avoid-the-machine-gun-effect-using-round-robin-sampling-632302 )

**Comment by Lykdraft** [comment]
Pushing this. Multisample capability.
